### PR TITLE
New runs table column view

### PR DESCRIPTION
**Pull request by Zangr**
## What changes are proposed in this pull request?

This PR implements a new runs table column view based on `ag-grid` with a series of features as listed below:
- Nested runs
- Serverside sorting
- Load more runs
- Column custom selection with a checkable tree dropdown
- Column resizing
- Column reordering
- Column pinning
- Column highlighting
- Column toggling
- Row selection batching with `Shift + Click`
- Cell `-` placeholder
- Table loading overlay
- Wire up the old compact view with the column selection tree dropdown

new-multi-column-view

## How is this patch tested?
Manual, Unit tests.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

This PR implements a new runs table column view based on `ag-grid` with a series of features 

### What component(s) does this PR affect?

- [x] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by smurching — COMMENTED** [review]
Thanks for this PR! As per offline discussion / review - the UX & implementation largely looks good. The wider-sort area for columns seems to be working, but causing columns to jump around gif. Maybe we can defer the wider-sort area to a follow-up PR? LGTM otherwise

**Comment by Zangr** [comment]
> The wider-sort area for columns seems to be working, but causing columns to jump around


The "jump around" is actually expected for here because sorting is server-side based and the availability of metrics/params/tags before and after sort varies.

File: mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    metricKeyList: [],
+    visibleTagKeyList: [],
+  };
+
+  state = {
+    menuVisible: false,
+  };
+
+  handleCheck = (checkedKeys, allKeys) => {
+    console.log('checkedKeys', checkedKeys);
```
Comment (harupy) [inline mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: This line should be removed.

**Comment by harupy** [comment]
The metrics checkbox can't be unchecked. Is this a bug or am I missing something?
bug

File: mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+function getCategorizedUncheckedKeys(checkedKeys, allKeys) {
+  const uncheckedKeys = _.difference(allKeys, checkedKeys);
+  const result = {
+    [ColumnTypes.ATTRIBUTES]: [],
+    [ColumnTypes.PARAMS]: [],
+    [ColumnTypes.METRICS]: [],
+    [ColumnTypes.TAGS]: [],
+  };
+  uncheckedKeys.forEach((key) => {
+    const [columnType, rawKey] = key.split('-');
```
Comment (harupy) [inline mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: This line doesn't split `key` properly when a metric (or parameter) name contains `-`.

Comment (harupy) [reply mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: Tested the code below and it worked.

```diff
+    const [columnType, rawKey] = key.split('-');
-    const [columnType, rawKey] = key.split(/-(.+)/);
```

Ref. : https://stackoverflow.com/questions/4607745/split-string-only-on-first-instance-of-specified-character

Comment (Zangr) [reply mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: Fixed

File: mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js (hunk 1 above)
Comment (Zangr) [reply mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: Removed

**Review by Zangr — COMMENTED** [review]
Addressed comments, disabled column drag/drop for now since we have not implemented persistence of column ordering. Also, added run status display for the new view.

File: mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js (hunk 1 above)
Comment (harupy) [reply mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: Maybe we should let the linter detect `console.log`.

Comment (Zangr) [reply mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: I think it's OK to leave it open for now. It's used a lot during development phase, we probably don't want running a lint fix to auto remove them.

Comment (harupy) [reply mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js]: Agreed.

**Review by smurching — APPROVED** [review]
LGTM thanks @Zangr! Let's get this in & iterate on it in follow-ups if needed, thanks for the huge UX improvement :)

**Pull request merged**